docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+Fedora will be its base as it is for the SDK.
+
+The image will include the following:
+- the docker CLI
+- cargo make
+- buildsys, pubsys, testsys, etc.
+- Scripts accessed by cargo make, such as `docker-go`.
+- `Makefile.toml`
+
+It is within this container that Twoliter will invoke `cargo make` to initiate a build.
+
+### SDK Changes
+
+The existing build system expects access to the mounted core project git tree for certain things during the build.
+These will not be available and need to be added to the Bottlerocket SDK:
+- Scripts used by `docker build` commands such as `rpm2img`, `rpm2kmodkit` and `partplanner`.
+- RPM macros.
+
+### Docker Domain Socket
+
+Twoliter will mount the host's docker socket (i.e. `/var/run/docker.sock`).
+This means that `docker` commands inside the Twoliter container environment will be interacting with the host's daemon.
+This presents certain challenges because directories need to be mounted from "within" this container.
+In particular:
+
+* The Twoliter container process needs to have `docker` group permissions.
+* The Twoliter container process needs to have permissions to write in certain host directories.
+
+We can solve this by introspecting the Twoliter CLI's process to get the UID and GID it is invoked with.
+We make the assumption that the user invoking the CLI has the permissions to interact with docker and to write in the desired host directories.
+We then pass `--user "uid:gid"` to docker when starting the Twoliter container.
+We can also allow the Twoliter command line to accept uid and gid arguments to override this behavior.
+
+An additional problem to solve is that the paths to mounted directories need to be absolutely the same on the host as they are in the Twoliter container.
+This will not be problem unless the host build or sourcecode directory is in some wildly unusual place, like `/etc/docker` or `/usr/bin`.
+We can prevent this by making a certain set of top-level directories off-limits on the host for the directories we need to mount.
+An error message will describe the problem instead of running the container and failing due to occluded directories.
+
+## Twoliter Projects
+
+Twoliter will have the concept of a project.
+A project can contain multiple variant definitions, multiple kit definitions, and multiple packages.
+
+A project will have a top level `Twoliter.toml` file (and possibly a `Twoliter.lock` file).
+When `twoliter` commands are invoked, the tool will search for the `Twoliter.toml` file in `.`.
+If it doesn't find it, it will search in `..`, then `../..`, etc.
+This will allow the tool to work from any subdirectory within the project (and is inspired by `cargo`).
+
+`Twoliter.toml` will contain global project settings. Here is a sample `Twoliter.toml` file.
+
+```toml
+# This let's twoliter know if it is reading an older version of the Twoliter.toml
+# schema, or if the directory structure of the project is from an older version.
+twoliter-project-version = 1
```

Comment:
I'm struggling with what to call this as it conflates with `project-version` below.